src/org/traccar/protocol/WondexFrameDecoder.java

```diff
@@ -36,17 +38,17 @@ protected Object decode(
         if (buf.getUnsignedByte(buf.readerIndex()) == 0xD0) {
 
             // Send response
-            ChannelBuffer frame = buf.readBytes(KEEP_ALIVE_LENGTH);
-            if (channel != null) {
-                channel.write(frame);
+            ByteBuf frame = buf.readBytes(KEEP_ALIVE_LENGTH);
+            if (ctx != null && ctx.channel() != null) {
+                ctx.channel().writeAndFlush(new NetworkMessage(frame, ctx.channel().remoteAddress()));
             }
             return frame;
 
         } else {
 
-            int index = buf.indexOf(buf.readerIndex(), buf.writerIndex(), new StringFinder("\r\n"));
+            int index = new AsciiString(buf.array(), buf.readerIndex(), buf.writerIndex(), false).indexOf("\r\n");
```

Comment:
Just a note here. I like this option if buffer is a string, but in some cases it can be a mix of characters and binary data. In that case I like my solution better.